### PR TITLE
custom-diff: x/gov

### PR DESCRIPTION
**Pull request by dongsam**
- [feat: add additional voting powers hook on tally](https://github.com/cosmosquad-labs/cosmos-sdk/commit/d76f9225b259c4705dd4c368c4c0a4b6f8f673cd)

Hook, interface, and logic were added to allow other modules to set additional voting power other than delegation voting power together.

### CODE REVIEW

File: x/gov/keeper/hooks_test.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type MockGovHooksReceiver struct {
 	AfterProposalVoteValid              bool
 	AfterProposalFailedMinDepositValid  bool
 	AfterProposalVotingPeriodEndedValid bool
+	GetAdditionalVotingPowersValid      bool
```
Comment (hallazzang) [inline x/gov/keeper/hooks_test.go:26]:
```suggestion
	SetAdditionalVotingPowersValid      bool
```

File: x/gov/spec/02_state.md [hunk 2]
```diff
@@ -159,8 +159,14 @@ And the pseudocode for the `ProposalProcessingQueue`:
...
-      voterIterator = rangeQuery(Governance, <proposalID|'addresses'>) //return all the addresses that voted on the proposal
-      for each (voterAddress, vote) in voterIterator
+      voters = rangeQuery(Governance, <proposalID|'addresses'>) //return all the addresses that voted on the proposal
+	  // set additional voting powers by hooking other modules
+	  additionalVotingPowersMap = SetAdditionalVotingPowers(voters)
+      for each (voterAddress, vote) in voters
+        for each (validator, votingPower) in additionalVotingPowersMap[voterAddress]
+          tmpValMap(validator).Minus += votingPower.Shares
+          proposal.updateTally(vote, votingPower.Shares)
+
```
Comment (hallazzang) [inline x/gov/spec/02_state.md:169]: I think indentation is not consistent.

File: x/gov/types/vote.go [hunk 3]
```diff
@@ -24,6 +24,9 @@ func (v Vote) String() string {
 // Votes is a collection of Vote objects
 type Votes []Vote
 
+// AdditionalVotingPowers is additional votingPower map by validators by voters
```
Comment (hallazzang) [inline x/gov/types/vote.go:27]: Need more detailed description about keys, something like:
```go
// validator addr => (voter addr => additional voting power)
```

Comment (dongsam) [reply x/gov/types/vote.go:27]: actually, `voter addr => (validator addr => additional voting power)`, Thanks, Let me add

File: x/gov/spec/02_state.md (hunk 2 above)
Comment (dongsam) [reply x/gov/spec/02_state.md:169]: Thanks, Let me fix

**Pull request closed without merging**